### PR TITLE
Feat/scoped storage

### LINKED ISSUE #490: Feature Request: manage files in the "Android/data" folder on Android 11
As the title says. More information in this topic on Reddit: https://www.reddit.com/r/Android/comments/j3zgmm/managing_files_in_the_androiddata_folder_on/

### PR DESCRIPTION
**Pull request by KryptKode**
**Notes**
- update targetSdkVersion to 30
- request for `MANAGE_EXTERNAL_STORAGE` permission for devices Android 11 and above. 
- handle access to  `Android/[data|obb]` on the primary volume  with the Storage Access Framework due to scoped storage restrictions on devices running on Android 11 and above (>= Android 11)
- closes #490 
- [this PR](https://github.com/SimpleMobileTools/Simple-Commons/pull/1202) should be merged before the project can be built due to the updated commons dependency

### CODE REVIEW

**Comment by KryptKode** [comment]
@tibbi  Do we need to display an explanation for the need for the `MANAGE_EXTERNAL_STORAGE` permission

File: settings.gradle [hunk 1]
```diff
@@ -1 +1,3 @@
 include ':app'
+include ':commons'
+project(":commons").projectDir = new File("/Users/darthpaul/StudioProjects/Simple-Commons/commons")
```
Comment (tibbi) [inline settings.gradle:3]: guess this shouldnt be here. Try using some GUI for viewing changes before commiting

**Comment by tibbi** [comment]
> @tibbi Do we need to display an explanation for the need for the `MANAGE_EXTERNAL_STORAGE` permission

I dont think we do. But we need to redirect the user to the device settings to allow the management of all files, no? I haven't checked the code yet, not sure if you are doing it already.

**Comment by KryptKode** [comment]
> > @tibbi Do we need to display an explanation for the need for the `MANAGE_EXTERNAL_STORAGE` permission
> 
> I dont think we do. But we need to redirect the user to the device settings to allow the management of all files, no? I haven't checked the code yet, not sure if you are doing it already.

Yes, the check is there. I will upload a screenshot later.

File: app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/MainActivity.kt [hunk 2]
```diff
@@ -300,6 +306,43 @@ class MainActivity : SimpleActivity() {
...
+            if (isRPlus()) {
+                isAskingPermissions = true
+                actionOnPermission = callback
+                try {
+                    val intent = Intent(Settings.ACTION_MANAGE_APP_ALL_FILES_ACCESS_PERMISSION)
+                    intent.addCategory("android.intent.category.DEFAULT")
+                    intent.data = Uri.parse("package:$packageName")
+                    startActivityForResult(intent, MANAGE_STORAGE_RC)
+                } catch (e: Exception) {
+                    e.printStackTrace()
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/MainActivity.kt:323]: lets not print errors

**Comment by tibbi** [comment]
this is a massive change too, maybe upgrading to SDK 30 would have been enough as the first step :/ Will have to do some testing too. Not just fresh installs but upgrading from the old versions too.

**Comment by KryptKode** [comment]
> this is a massive change too, maybe upgrading to SDK 30 would have been enough as the first step :/ Will have to do some testing too. Not just fresh installs but upgrading from the old versions too.

Upgrading the app on Android 11 should work like Android 10 if we set the `preserveLegacyExternalStorage` attribute  to `true` in the `AndroidManifest`

**Comment by tibbi** [comment]
oh, we definitely need an explanation before the redirection, not just throw the user there automatically. I didnt know you ask that.

**Comment by tibbi** [comment]
added a new string that should be used at the confirmation (explanation) dialog

**Comment by tibbi** [comment]
@KryptKode have you tested these with an Android 11 device with an SD card? Both an upgrade from the old app version to this new one, and also a fresh new install? I just noticed that my SD card device is on Android 10 only :/

**Comment by tibbi** [comment]
plus as a note to myself, the extra permission text mentions an SD card and the image too, will have to update it

**Comment by KryptKode** [comment]
> @KryptKode have you tested these with an Android 11 device with an SD card? Both an upgrade from the old app version to this new one, and also a fresh new install? I just noticed that my SD card device is on Android 10 only and I have no adapter for USB C :/

I only tested this on an emulator. The Android 11 I have does not support SD cards. I can get an OTG though to test.

**Comment by tibbi** [comment]
but you tested Android 11 SD card handling on the emulator? Or what is the chance that something broke there?

**Comment by KryptKode** [comment]
> but you tested Android 11 SD card handling on the emulator? Or what is the chance that something broke there?

Chances are slim but we should test on real devices too.

**Comment by tibbi** [comment]
I added a new string into Simple Commons that should be used there, but it needs a new dialog as the image should be removed too. I dont want to mix both things into one dialog as it already toggles between SD and OTG.

**Comment by KryptKode** [comment]
Do you mean for prompting users to select Android/[data|obb]?

**Comment by tibbi** [comment]
yes, using a new dialog instead of WritePermissionDialog

**Comment by tibbi** [comment]
meh I messed up some branches, will accept this and fix the dialogs in master

**Pull request merged**